main.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+module Enumerable #:nodoc:
+  def my_each
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each { |x| new_array << x if yield(x) }
+    new_array
+  end
+
+  def my_all?
+    return true if size < 1
+    return false if block_given? && nil? || !block_given?
+
+    my_each { |x| return false unless yield(x) }
+    true
+  end
+
+  def my_any?
+    return false if size < 1
+    return true unless block_given?
+
+    my_each { |x| return true if yield(x) }
+    false
+  end
+
+  def my_none?(arg = nil)
+    if block_given?
+      my_each { |x| return false if yield(x) }
+    elsif !arg.nil?
+      if arg.is_a?(Regexp)
+        my_each { |x| return false if pattern =~ x.to_s }
+      else
+        my_each { |x| return false if x.is_a?(pattern) }
+      end
+    else
+      my_each { |x| return false if x }
+    end
+    true
+  end
```

Comment:
* [ ] my_none? should be able to accept **literals, classes i.e Integer), regex expressions** as argument. Yours crashes when doing so :crying_cat_face: